### DISCUSSION TITLE
Simconnect and the Arduino board update speed?

### DISCUSSION THREAD

**Discussion by Thick8** [MobiFlight/MobiFlight-Connector#805] [Q&A] [answered]
Hi All,
I am using Mobiflight to run my FFB rudder pedals. I've lowered the analog read and average. This helped reduce the lag. How do I increase the frequency that the Simconnect info is sent to the Arduino? My hope is that I can decrease the lag more. 
I am not, in any way, shape, or form, a coder. So please explain it as you would to the noobyist of noobs.
Thanks.
John

**Comment by Thick8** [comment]
Would I up the baud rate?

**Comment by tigert** [comment]
This is a good question. I think the update speed depends on MobiFlight Connector reading and writing the arduino pin states, this is not the same as the baud rate.

I have been thinking a bit of the same, but Mobiflight architecture was not designed to be realtime responsive with large amounts of data, as it is meant to track button and switch events and to drive warning lights and other outputs based on simulator events.

That said, for example 7-segment leds do update quite fast even when you write the value to the sim and read it back, having a round trip, but I am not sure if it is fast enough still for force feedback? I have not gone that far with my experiments yet. I need to build a stick or pedals with potentiometer feedback and then just try how those work if I try to create a centering spring force for example, if it will be hopelessly laggy.

What hardware are you using to drive the force? I experimented a bit with a H-bridge and brushed DC motor, but I am not at all sure if it would work with mobiflight update loop. 

I can see your pain point, also not being that much of a programmer, so being able to run the logic in mobiflight would be cool, but I am not sure if this is going to work in practice.

**Comment by DocMoebiuz** [comment, marked as answer]
values are sent to the arduinos every time they change.

values are read from the sim in different ways depending on the flight sim.

fsuipc connection requests the data with every loop through the output configs.

sim connect receives the data as soon as it changes.

xplane 10 times per second.

However, the current values are only evaluated when an individual config item is executed because there are transforms and other manipulation possible.

Bottom line, if you want to increase the responsiveness then you have the execution speed slider in the settings menu. the best value is to perform updates every 50ms or 20 times per second.

**Comment by Thick8** [comment]
Thanks for your input guys.
I using a 24 volt 775 motor and BTS7960 driver. I've come to realize that a big part of the issue is the amount of free-play in the Saitek rudder pedals. The set screw on the main gear is also backing out and allowing it to rotate slightly on the motor shaft. I just finished a redesign using 8020 aluminum that should take care of the slop.
Mobiflight is reading the attached potentiometers along with a few Simconnect variable for wind velocity and direction. Those variables are used as transform values in the motor direction and force outputs. Overall it works pretty good. the pedals follow AP, the center point changes with crosswind, and they go a little nuts in high winds with variable gusts. 
I'm going to run some more tests once the new rudder pedals are up and running.
